### DISCUSSION TITLE
Can I place the title above a figure in a MS Word rendered document?

### DISCUSSION THREAD

**Discussion by jenineharris** [quarto-dev/quarto-cli#3467] [Q&A] [answered]
I am using the "fig-cap-location: top" as part of the YAML in an HTML rendered document in Quarto successfully but when I try to render the same document to MS Word, the figure caption is below the figure (also sub captions are below). I have added the same argument as a code chunk argument. Here is a reproducible example to paste into a blank qmd file:

---
title: "reproducible example"
format: docx
fig-cap-location: top
editor: visual
---

```{r}
#| echo: false
#| message: false
#| warning: false
#| fig-cap: Title that goes across 2 figures, preferably at the top. 
#| fig-subcap: 
#|  - "Cars speed density plot"
#|  - "Cars distance density plot"
#| label: fig-cars
#| layout-ncol: 2

# I've also tried it with the fig-cap-location: top option as a chunk option
# The help documentation does not mention Word documents so maybe this is unavailable so far?
# URL help docs: https://quarto.org/docs/authoring/figures.html

library(tidyverse)

cars %>% 
  ggplot(aes(x = speed)) +
  geom_density()

cars %>% 
  ggplot(aes(x = dist)) +
  geom_density()

```

**Comment by mcanouil** [comment]
Oh, you are using Word.
Unfortunately, this is defined in the default Word template.
You have to define your own by modifying the style in it.
See <https://quarto.org/docs/output-formats/ms-word-templates.html>.

**Reply by jenineharris** [reply]
Thank you, I am using a Word template. The table caption placement is working fine with my current template (moves from below to above in the rendered Word docx using the *tab-cap-location: top* in the code chunk options). 

The problem with the figure captions is they are rendering in Word as separate from the figures themselves (rather than as a caption connected to the figure as a property of the figure), so using the option in word style pane to move them above doesn't work because they are not actually directly linked to the figures for some reason, they are separate text unrelated to the figures but placed just below. See rendered document attached. 
[Formatted-reproducible-project-word.docx](https://github.com/quarto-dev/quarto-cli/files/10098562/Formatted-reproducible-project-word.docx)

Thanks!

**Reply by mcanouil** [reply]
Let me rephrase, the only way to change the caption location is from the docx document either using some post-render script, tweak the template, or something else.
`fig-cap-location` is only valid for `html` and `pdf` formats. This is why the option is not described in <https://quarto.org/docs/reference/formats/docx.html>.

The issue is mostly in Pandoc support to docx unfortunately, i.e., if you use Pandoc directly, the generated Docx does not detect the figure/caption pair as it does when manually inserting a caption, where the layout can be modified.

**Reply by jenineharris** [reply]
Ok, thanks. I was hopeful given that the table caption worked even though it is also not in the quarto help docs for docx. Hopefully a future version will have this! I actually learned quite a bit about what is possible in MS Word through trying to figure this out, so I'll take that as a win for now. Appreciate your time and the information. 🙂

**Reply by mcanouil** [reply]
Maybe, something in Quarto can be done to insert the caption above and not rely on Pandoc or the Docx template, _i.e._, there is a Lua filter for docx (<https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/filters/layout/docx.lua>) and for table there is some code for caption position (<https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/filters/layout/table.lua>).

**Reply by mcanouil** [reply]
@dragonstyle I saw you contributed some code to the Lua scripts regarding captions for `docx` format, maybe you have some thoughts?

**Reply by mcanouil** [reply]
@jenineharris Could you change the category from "general" to "Q&A"?[^1] This way, if an answer/solution appears (watch for #2221), you can mark it which will let others know it has an answer.

[^1]: Only possible using the desktop view of GitHub website.

**Reply by jenineharris** [reply]
Great, thanks, that seems promising and I'll take a look. Unfortunately, I don't know anything about Lua or how to modify/add Pandoc features so I'll have to put this on my list for after the semester ends. Thanks again for the suggestions and your time. I'm going to see if I can hack it by putting the figures in some sort of table and then give up for now 😆.

**Reply by jenineharris** [reply]
> @jenineharris Could you change the category from "general" to "Q&A"?[1](#user-content-fn-1-147cb29f07a43763924c70d67b52e352) This way, if an answer/solution appears (watch for #2221), you can mark it which will let others know it has an answer.
> 
> ## Footnotes
> 1. Only possible using the desktop view of GitHub website. [↩](#user-content-fnref-1-147cb29f07a43763924c70d67b52e352)

Will do! thanks.

**Reply by mcanouil** [reply]
> I'll have to put this on my list for after the semester ends

Hopefully, this could be fixed internally by Quarto team before that.

**Reply by matanhakim** [reply]
Hi all!
This issue is of value to me as well.
Has something been changed or added by now?
Thanks a lot for your work.

**Reply by cscheid** [reply]
The comment below yours has a link to the github issue. When the issue is fixed, the github issue will be closed :)

**Comment by mcanouil** [comment, marked as answer]
FYI, there is an open issue for this: #2221

**Comment by Semuel12** [comment]
(spam removed).

**Reply by jenineharris** [reply]
Thanks for the reply. I'm not sure why you are presuming what sort of document I need.

**Reply by mcanouil** [reply]
And a link to a clearly off topic website ...
Better to disregard this as even if it is not, it definitely looks like spam.

**Reply by jenineharris** [reply]
I didn't click on the link 😬. Thanks.

**Reply by mcanouil** [reply]
I "hovered" it, "nyfleets" or something like that ...
Why hide URL if you want to make a point with an example ...

**Reply by mcanouil** [reply]
@dragonstyle @cscheid hey guys, can you do something? At least remove the website link?

FYI, I reported the account and post, for spam and possibly phishing attempt.
The account is as old as it's message here.
